packages/editor/src/utils/media-upload/media-upload.js

```diff
@@ -68,7 +68,11 @@ export function mediaUpload( {
 
 	// Allowed type specified by consumer
 	const isAllowedType = ( fileType ) => {
-		return ( allowedType === '*' ) || startsWith( fileType, `${ allowedType }/` );
+		return ( allowedType === '*' ) ||
+			some(
+				( allowedType || '' ).split( ',' ),
```

Comment:
Why do we have `allowedType` as this weird string thing, when we already have sufficient types in JavaScript to represent them more semantically:

- If all types are supported, why not just not pass the property rather than a `'*'` string ?
- If multiple types are supported, why not just pass an array of mime type strings?

Reply:
Hi @aduth thank you for the feedback I agree it is preferable to use an array, although this change involves a refactor and deprecations. I updated the code to use this version, I think things should work as before and we added deprecation messages for all API changes.